source/byte_buf.c

```diff
@@ -57,6 +57,46 @@ void aws_byte_buf_secure_clean_up(struct aws_byte_buf *buf) {
     aws_byte_buf_clean_up(buf);
 }
 
+bool aws_byte_buf_eq(const struct aws_byte_buf *a, const struct aws_byte_buf *b) {
+    if (!a || !b) {
+        return (a == b);
+    }
+
+    if (a->len != b->len) {
+        return false;
+    }
+
+    if (!a->buffer || !b->buffer) {
+        return (a->buffer == b->buffer);
+    }
+
+    return !memcmp(a->buffer, b->buffer, a->len);
+}
+
+int aws_byte_buf_copy(struct aws_allocator *allocator, struct aws_byte_buf *dest, const struct aws_byte_buf *src) {
```

Comment:
I would call this `aws_byte_buf_dup` - as in `strdup` - to clarify that it's allocating the memory instead of copying into an existing buffer.